src/noit_metric_director.c

```diff
@@ -577,3 +586,7 @@ noit_metric_director_get_messages_distributed() {
  return ck_pr_load_64(&number_of_messages_distributed);
 }
 
+void
+noit_metric_director_drop_before(double t) {
+  drop_before_threshold_ms = (uint64_t) t*1000;
```

Comment:
You want `(uint64_t) (t*1000.0)`

As is, t is converted to a uint64_t before multiplying and will always look milliseconds.

Reply:
Removed the cast.